stl/inc/format

```diff
@@ -1045,7 +1048,7 @@ private:
         const auto _Length      = _Get_format_arg_type_storage_size<_CharType>(_Arg_type);
 
         _CSTD memcpy(_Storage + _Index_length + _Store_index, _STD addressof(_Val), _Length);
-        _Index_array[_Arg_index]._Type = _Arg_type;
+        _Index_array[_Arg_index]._Type = static_cast<_Index_type::_Index_type>(_Arg_type);

```

Comment:
can we just attach a member function to _Index_type that does this?

Reply:
I originally did that, but it's only done once to assign and once to access, so it felt kind of needlessly wordy. I still can, I just wasn't a fan of it.

Reply:
I would rather not. If it is rare, we can avoid the debug hit